$OpenBSD: patch-scripts_util_ssvnc_tcl,v 1.2 2009/10/31 16:03:01 sthen Exp $

type sometimes returns "xx is a tracked alias for /foo/bar" which
isn't handled. whence is sufficient and more reliable.

--- scripts/util/ssvnc.tcl.orig	Wed Oct 28 15:50:07 2009
+++ scripts/util/ssvnc.tcl	Sat Oct 31 15:47:11 2009
@@ -16026,7 +16026,7 @@ proc check_zeroconf_browse {} {
 		return 0;
 	}
 	set p ""
-	set r [catch {set p [exec /bin/sh -c {type avahi-browse}]}]
+	set r [catch {set p [exec /bin/sh -c {whence avahi-browse}]}]
 	if {$r == 0} {
 		regsub {^.* is  *} $p "" p
 		regsub -all {[ \t\n\r]} $p "" p
@@ -16036,7 +16036,7 @@ proc check_zeroconf_browse {} {
 		}
 	}
 	set p ""
-	set r [catch {set p [exec /bin/sh -c {type dns-sd}]}]
+	set r [catch {set p [exec /bin/sh -c {whence dns-sd}]}]
 	if {$r == 0} {
 		regsub {^.* is  *} $p "" p
 		regsub -all {[ \t\n\r]} $p "" p
@@ -16051,7 +16051,7 @@ proc check_zeroconf_browse {} {
 		}
 	}
 	set p ""
-	set r [catch {set p [exec /bin/sh -c {type mDNS}]}]
+	set r [catch {set p [exec /bin/sh -c {whence mDNS}]}]
 	if {$r == 0} {
 		regsub {^.* is  *} $p "" p
 		regsub -all {[ \t\n\r]} $p "" p
